src/features/tokens/TokenSelectField.tsx

```diff
@@ -26,23 +29,34 @@ export function TokenSelectField({ name, disabled, setIsNft }: Props) {
     const tokensWithRoute = warpCore.getTokensForRoute(origin, destination);
     let newFieldValue: number | undefined;
     let newIsAutomatic: boolean;
+
+    // Use token from persistance
+    if (query.token && typeof query.token === 'string') {
```

Comment:
Will this re-trigger if a user chooses a different chain but there's still an unrelated token value in the URL